Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -400,10 +347,139 @@ public void JumpMovement( Vector3 inputVec )
 		}
 	}
 
-	protected void ChangeState( ActorStates toState )
+	public void StartGlide()
+	{
+		if ( !_isOnGround )
+		{
+			ChangeState( ActorStates.Gliding );
+
+			rigidbody.useGravity = false;
+
+			if ( actor.animator )
+			{
+				actor.animator.SetBool( "isJumping", true );
+			}
+		}
+	}
+
+	public void EndGlide()
+	{
+		rigidbody.useGravity = true;
+
+		if ( actor.animator )
+		{
+			actor.animator.SetBool( "isJumping", false );
+		}
+	}
+
+	public void GlideMovement( Vector3 inputVec )
+	{
+		_moveVec = inputVec * _glideHorizontalSpeed;
+		_moveVec.y = -_glideDescentRate;
+
+		rigidbody.velocity = _moveVec;
+	}
+
+	public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
+	{
+		_stateMethodMap.Add( state, method );
+	}
+
+	void ConstrainBumper()
+	{
+		_bumperTransform.position = transform.position;
+		_bumperRigidbody.velocity = rigidbody.velocity;
+	}
+
+	void FollowBumper()
+	{
+		if ( !IsInState( ActorStates.Climbing ) )
+		{
+			Vector3 constrainedPos = transform.position;
+			constrainedPos.x = _bumperTransform.position.x;
+			constrainedPos.z = _bumperTransform.position.z;
+			transform.position = constrainedPos;
+		}
+	}
+
+	void MoveAtSpeed( Vector3 inVec, float moveSpeed )
+	{
+		_inputVec = inVec;
+		rigidbody.useGravity = true;
+
+		_moveVec = inVec * moveSpeed * _moveSpeedMod;
+		_moveVec.y = rigidbody.velocity.y;
+
+		rigidbody.velocity = _moveVec;
+
+		if ( actor.animator )
+		{
+			actor.animator.SetBool( "isMoving", true );
+		}
+	}
+
+	/**
+	 * Provide resonable default values for the state methods.
+	 * 
+	 * If the state method map has already been configured
+	 * before this is called, the methods will not be overriden.
+	 */
+	void InitializeStateMethodMap()
+	{
+		foreach ( ActorStates state in System.Enum.GetValues( typeof( ActorStates ) ) )
+		{
+			if ( !_stateMethodMap.ContainsKey( state ) )
+			{
+				_stateMethodMap.Add( state, delegate() { } );
+			}
+		}
+	}
+
+	void ChangeState( ActorStates toState )
 	{
 		_currentState = toState;
-		CurrentStateMethod = stateMethodMap[_currentState];
+		CurrentStateMethod = _stateMethodMap[_currentState];
+	}
+
+	void StartClimbing( Collider col )
+	{
+		_climbSurface = col.transform;
+		_climbTag = _climbSurface.GetComponent<ClimbableTag>();
+
+		if ( Vector3.Dot( _climbSurface.forward, ( transform.position - _climbSurface.position ) ) > 0.0f )
+		{
+			_climbSurfaceNormal = -_climbSurface.forward;
+			_climbSurfaceRight = -_climbSurface.right;
+		}
+		else
+		{
+			_climbSurfaceNormal = _climbSurface.forward;
+			_climbSurfaceRight = _climbSurface.right;
+			Debug.LogWarning( "Climb volume is facing into wall." );
+		}
+
+		if ( Vector3.Dot( Vector3.up, _climbSurface.up ) > 0.0f )
+		{
+			_climbSurfaceUp = _climbSurface.up;
+		}
+		else
+		{
+			_climbSurfaceUp = -_climbSurface.up;
+			Debug.LogWarning( "Climb volume is upside down." );
```

Comment:
He chose this way so we'll know it's wrong and should be fixed but it isn't game breaking priority like an error would be.